packages/core/src/server/middlewares/settings-and-store.ts

```diff
@@ -0,0 +1,64 @@
+import { Middleware, Next } from "koa";
+import { Context } from "@frontity/types";
+import { getSettings } from "@frontity/file-settings";
+import createStore from "../store";
+import { appComponent } from "./capabilities/app-component";
+import { renderMethod } from "./capabilities/render-method";
+import { template } from "./capabilities/template";
+
+/**
+ * Setup the settings and defines the state store.
+ *
+ * @param packages - The frontity packages.
+ *
+ * @returns The middleware function.
+ */
+export const settingsAndStore = (packages) => async (
+  ctx: Context,
+  next: Next
+): Promise<Middleware> => {
+  // Get settings.
+  const settings = (ctx.state.settings = await getSettings({
+    url: ctx.href,
+    name: ctx.query.frontity_name,
+  }));
+
+  // Create the store.
+  const store = (ctx.state.store = createStore({
+    settings: settings,
+    packages,
+    url: ctx.URL,
+  }));
+
+  // Apply the capabilities to the store.libraries.frontity namespce.
+  appComponent(store.libraries.frontity, ctx);
+
+  // Setup the render method.
+  renderMethod(store.libraries.frontity);
+
+  // Define the default template.
+  template(store.libraries.frontity);
+
+  // Run init actions.
+  await Promise.all(
+    Object.values(store.actions).map(({ init }) => {
+      if (init) return init();
+    })
+  );
+
+  // Run beforeSSR actions.
+  await Promise.all(
+    Object.values(store.actions).map(({ beforeSSR }) => {
+      if (beforeSSR) return beforeSSR({ ctx, libraries: store.libraries });
```

Comment:
That's a good question! Yeah it was a bit confusing that part, since it can accept a function or a curried function.
So, if one uses the single function indeed the `store` is sent but for the curried one, you get only the `ctx` which is expected.

Reverted the change. 👍  Awesome find! 🙌